lib/tar.ml

```diff
@@ -859,3 +861,41 @@ let fold f init =
     | Error `Eof -> return (Ok acc)
     | Error `Fatal _ as e -> return e in
   go (decode_state ()) init
+
+let rec writev = function
+  | [] -> return (Ok ())
+  | x :: r ->
+    let* () = write x in
+    writev r
+
+let rec pipe stream =
+  let* block = stream () in
+  match block with
+  | Some str -> let* () = writev [ str ] in pipe stream
+  | None -> return (Ok ())
+
+type ('err, 't) content = unit -> (string option, 'err, 't) t
+type ('err, 't) entry = Header.compatibility option * Header.t * ('err, 't) content
+type ('err, 't) entries = unit -> (('err, 't) entry option, 'err, 't) t
```

Comment:
I don't have strong opinion but we should check which version of OCaml we require for `Seq.t`.

Reply:
Actually, `Seq.t` appeared mainly since 4.14. `ocaml-tar` is used by some people and I think we still support, at least, OCaml 4.13. So, I prefer to keep this type instead of `Seq.t`.